blockchain/state_manager/src/utils.rs

```diff
@@ -100,3 +109,270 @@ where
+    tipset: &Tipset,
+    address: &Address,
+) -> Result<Deadlines, Error>
+where
+    DB: BlockStore,
+{
+    let miner_actor_state: miner::State = state_manager
+        .load_actor_state(&address, &tipset.parent_state())
+        .map_err(|err| {
+            Error::State(format!(
+                "(get miner deadlines) failed to load miner actor state: {:}",
+                err
+            ))
+        })?;
+    miner_actor_state
+        .load_deadlines(&*state_manager.get_block_store())
+        .map_err(|err| {
+            Error::State(format!(
+                "(get_miner_deadlines) could not load deadlines: {:}",
+                err
+            ))
+        })
+}
+
+pub fn get_miner_faults<DB>(
+    state_manager: &StateManager<DB>,
+    tipset: &Tipset,
+    address: &Address,
+) -> Result<BitField, Error>
+where
+    DB: BlockStore,
+{
+    let miner_actor_state: miner::State = state_manager
+        .load_actor_state(&address, &tipset.parent_state())
+        .map_err(|err| {
+            Error::State(format!(
+                "(get miner faults) failed to load miner actor state: {:}",
+                err
+            ))
+        })?;
+    Ok(miner_actor_state.faults)
+}
+
+pub fn get_miner_recoveries<DB>(
+    state_manager: &StateManager<DB>,
+    tipset: &Tipset,
+    address: &Address,
+) -> Result<BitField, Error>
+where
+    DB: BlockStore,
+{
+    let miner_actor_state: miner::State = state_manager
+        .load_actor_state(&address, &tipset.parent_state())
+        .map_err(|err| {
+            Error::State(format!(
+                "(get miner recoveries) failed to load miner actor state: {:}",
+                err
+            ))
+        })?;
+    Ok(miner_actor_state.recoveries)
+}
+
+pub fn list_miner_actors<'a, DB>(
+    state_manager: &'a StateManager<DB>,
+    tipset: &'a Tipset,
+) -> Result<Vec<Address>, Error>
+where
+    DB: BlockStore,
+{
+    let power_actor_state: power::State = state_manager
+        .load_actor_state(&actor::STORAGE_POWER_ACTOR_ADDR, &tipset.parent_state())
+        .map_err(|err| {
+            Error::State(format!(
+                "(get sset) failed to load power actor state: {:}",
+                err
+            ))
+        })?;
+    let mut miners: Vec<Address> = Vec::new();
+    let block_store = &*state_manager.get_block_store();
+    let map = Hamt::load(&power_actor_state.claims, block_store)
+        .map_err(|err| Error::Other(err.to_string()))?;
+    map.for_each(|_: &String, k: String| -> Result<(), String> {
+        let address = Address::from_bytes(k.as_bytes()).map_err(|e| e.to_string())?;
+        miners.push(address);
+        Ok(())
+    })
+    .map_err(Error::Other)?;
+    Ok(miners)
+}
+
+pub fn get_power<DB>(
+    state_manager: &StateManager<DB>,
+    tipset: &Tipset,
+    address: Option<&Address>,
+) -> Result<(Option<Claim>, Claim), Error>
+where
+    DB: BlockStore,
+{
+    get_power_raw(state_manager, tipset.parent_state(), address)
+}
```

Comment:
I'm confused about why this is different from the one attached to the state manager?

Reply:
Yeah this was a mistake, shouldn't even be here.